src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] arr;
+    private T[] newArr;
+
+    public ArrayList() {
+        arr = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < arr.length) {
+            arr[size] = value;
+            size++;
+        } else {
+            resize();
+            arr[size] = value;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+        if (size >= arr.length) {
+            resize();
+        }
 
+        newArr = (T[]) new Object[arr.length];
```

Comment:
Use of 'newArr' is redundant here. You should use 'System.arraycopy()' directly on 'arr'.